beacon-chain/blockchain/optimistic_sync.go

```diff
@@ -174,3 +188,21 @@ func (s *Service) optimisticCandidateBlock(ctx context.Context, blk block.Beacon
 	}
 	return parentIsExecutionBlock, nil
 }
+
+// removeInvalidBlockAndState removes the invalid block and its corresponding state from the cache and DB.
+func (s *Service) removeInvalidBlockAndState(ctx context.Context, blkRoots [][32]byte) error {
+	for _, root := range blkRoots {
+		if err := s.cfg.StateGen.DeleteStateFromCaches(ctx, root); err != nil {
+			return err
+		}
+
+		// Delete block also deletes the state as well.
+		if err := s.cfg.BeaconDB.DeleteBlock(ctx, root); err != nil {
+			if err == kv.ErrDeleteJustifiedAndFinalized {
+				log.Panic("Invalid justified / finalized block in DB. Please resync from last weak subjectivity checkpoint")
+			}
```

Comment:
I don't think a panic is the right choice here. Can you surface the error further up the application?

Reply:
If this error happens, the node should be stopped. Are you suggesting surfacing this specific error further up the application to panic or to gracefully shut down the node?